nio/client/async_client.py

```diff
@@ -2928,3 +2933,60 @@ async def upload_filter(
         )
 
         return await self._send(UploadFilterResponse, method, path, data)
+
+    @logged_in
+    async def set_pushrule(
+        self,
+        scope: str,
+        kind: PushRuleKind,
+        rule_id: str,
+        before: Optional[str] = None,
+        after: Optional[str] = None,
+        actions: Sequence[PushAction] = (),
+        conditions: Optional[Sequence[PushCondition]] = None,
+        pattern: Optional[str] = None,
+    ) -> Union[SetPushRuleResponse, SetPushRuleError]:
+        """Create or modify an existing push rule.
+
+        Returns either a `SetPushRuleResponse` if the request was
+        successful or a `SetPushRuleError` if there was an error
+        with the request.
+
+        Args:
+            scope (str): The scope of this rule, e.g. ``"global"``.
+
+            kind (PushRuleKind): The kind of rule.
+
+            rule_id (str): The identifier of the rule. Must be unique
+                within its scope and kind.
+
+            before (Optional[str]): Position this rule before the one matching
+                the given rule ID.
+                The rule ID cannot belong to a predefined server rule.
+                ``before`` and ``after`` cannot be both specified.
+
+            after (Optional[str]): Position this rule after the one matching
+                the given rule ID.
+                The rule ID cannot belong to a predefined server rule.
+                ``before`` and ``after`` cannot be both specified.
+
+            actions (Sequence[PushAction]): Actions to perform when the
+                conditions for this rule are met. The given actions replace
+                the existing ones.
+
+            conditions (Sequence[PushCondition]): Event conditions that must
+                hold true for the rule to apply to that event.
+                A rule with no conditions always hold true.
+                Only applicable to ``underride`` and ``override`` rules.
+
+            pattern (Optional[str]): Glob-style pattern to match against
+                for the event's content.
+                Only applicable to ``content`` rules.
+        """
```

Comment:
Can we get an example for this, push rules are a bit of a complex thema and having an example that will get people going would be helpful.